From d929d29b2366e9178acb7597e1d346810a17752c Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Tue, 23 Jan 2018 09:49:31 +0200
Subject: [PATCH 3302/5242] MGS-3632-2: drm: imx: dcss: adjust DPR
 MAX_BYTES_PREQ depending on resolution

commit  cf567530abd9ed7d38d3ab08373b210d14d4933c from
https://source.codeaurora.org/external/imx/linux-imx.git

Current setting uses a 256 bytes/request for anything less than 1080p.
This works when DTRC is not involved. However, with DTRC, the
MAX_BYTES_PREQ needs to be fine tuned a little.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-dpr.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/imx/dcss/dcss-dpr.c b/drivers/gpu/imx/dcss/dcss-dpr.c
index 5d32bb3..749affd 100644
--- a/drivers/gpu/imx/dcss/dcss-dpr.c
+++ b/drivers/gpu/imx/dcss/dcss-dpr.c
@@ -274,8 +274,8 @@ void dcss_dpr_set_res(struct dcss_soc *dcss, int ch_num, u32 xres, u32 yres)
 		dcss_dpr_write(dpr, ch_num, pix_y_high,
 			       DCSS_DPR_FRAME_1P_PIX_Y_CTRL + plane * gap);
 
-		dcss_dpr_write(dpr,
-			       ch_num, xres < 1920 ? 2 : xres > 1920 ? 6 : 5,
+		dcss_dpr_write(dpr, ch_num, xres < 640 ? 3 :
+			       xres < 1280 ? 4 : xres < 3840 ? 5 : 6,
 			       DCSS_DPR_FRAME_1P_CTRL0 + plane * gap);
 	}
 }
-- 
1.7.9.5

